packages/core/types/src/types/core/attributes/blocks.ts

```diff
@@ -1,5 +1,75 @@
 import type { Attribute } from '..';
 
+interface TextInlineNode {
+  type: 'text';
+  text: string;
+  bold?: boolean;
+  italic?: boolean;
+  underline?: boolean;
+  strikethrough?: boolean;
+  code?: boolean;
+}
+
+interface LinkInlineNode {
+  type: 'link';
+  url: string;
+  children: TextInlineNode[];
+}
+
+interface ListItemInlineNode {
+  type: 'list-item';
+  children: DefaultInlineNode[];
+}
+
+type DefaultInlineNode = TextInlineNode | LinkInlineNode;
+type NonTextInlineNode = Exclude<DefaultInlineNode | ListItemInlineNode, TextInlineNode>;
```

Comment:
I would recommend you have a type with all inline nodes and just exclude from that instead to make it more maintanable :)